src/util.cc

```diff
@@ -423,21 +427,79 @@ const char* SpellcheckString(const char* text, ...) {
 }
 
 #ifdef _WIN32
+wstring Utf8ToWide(const string& u8_string) {
+  if (u8_string.empty())
+    return wstring();
+
+  size_t size_needed = MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
+                                           (int)u8_string.size(), NULL, 0);
```

Comment:
Typecast may result in loss of precision.

You need to assert that the size of the provided string (size_t) can actually be cast to an int safely.

Also, do not use c-style casts. Use static_cast<int>().

Reply:
static_cast is fine for now (`boost::numeric_cast` would be better, I really hope that it gets added to the standard some day).
A check seems overkill here (there are a lot of other things that will go wrong if we reach strings with a length of 2^31 characters).

Reply:
I agree, a boost::numeric_cast look-alike would be the best choice.

Perhaps someone will add something like that to Ninja while we wait for it to go into the standard lib.

I would strongly prefer that Ninja not make assumptions about things like string length.

At the very least, a call to assert() from <cassert> should only trigger with a debug build, right?

Reply:
Yes :)